### PR TITLE
Diff drive task space limits for velocity and acceleration

### PR DESCRIPTION
**Pull request by po1**
This is a rebased version of #16 with unit tests and even bug fixes.
This provides the requested feature in issue #12.

The open-loop acceleration limits will come in a separate pull request.

### CODE REVIEW

**Comment by efernandez** [comment]
:+1:

File: diff_drive_controller/include/diff_drive_controller/speed_limiter.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      bool has_acceleration_limits = false,
+      double min_velocity = 0.0,
+      double max_velocity = 0.0,
+      double min_acceleration = 0.0,
+      double max_acceleration = 0.0
+    );
+
+    /**
+     * @brief Limit the velocity and acceleration
+     * @param [out] v  Velocity
```
Comment (adolfo-rt) [inline diff_drive_controller/include/diff_drive_controller/speed_limiter.h]: Doc more clearly that v is in fact both an input and output parameter. This applies to other methods below as well

File: diff_drive_controller/test/diff_drive_common.launch [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  <!-- Load controller config -->
+  <rosparam command="load" file="$(find diff_drive_controller)/test/diffbot_controllers.yaml" />
+
+  <!-- Spawn controller -->
+  <node name="controller_spawner"
+        pkg="controller_manager" type="spawner" output="screen"
+        args="diffbot_controller" />
+
+  <!-- Rxplot monitoring -->
```
Comment (adolfo-rt) [inline diff_drive_controller/test/diff_drive_common.launch]: Pedantic: Rqt

File: diff_drive_controller/include/diff_drive_controller/speed_limiter.h (hunk 1 above)
Comment (efernandez) [inline diff_drive_controller/include/diff_drive_controller/speed_limiter.h]: Certainly, it should be:

<pre>
@param [in, out] v  Velocity
</pre>

which should be fine.

Also the units are missed. They must be:
[m/s]
[m/s]
[s]

**Pull request merged**